qemu/tests/virtio_serial_hotplug_max_chardevs.py

```diff
@@ -0,0 +1,108 @@
+    """
+    def run_serial_data_transfer():
+        """
+        Transfer data between two ports.
+        """
+
+        for serial_port in serials:
+            port_params = params.object_params(serial_port)
+            if not port_params['serial_type'].startswith('virtserial'):
+                continue
+            logging.info("transfer data with port %s" % serial_port)
+            params['file_transfer_serial_port'] = serial_port
+            transfer_data(params, vm, sender='both')
+
+    def run_bg_test():
+        """
+        Set the operation of transferring data as background
+        :return: return the background case thread if it's successful;
+                 else raise error
+        """
+        stress_thread = utils_misc.InterruptedThread(run_serial_data_transfer)
+        stress_thread.start()
+        if not utils_misc.wait_for(lambda: driver_in_use.check_bg_running(
+                vm, params), check_bg_timeout, 0, 1):
+            test.fail("Backgroud test is not alive!")
+        return stress_thread
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    os_type = params["os_type"]
+    check_bg_timeout = float(params.get('check_bg_timeout', 120))
+    num_chardev = int(params.get("numberic_chardev"))
+    num_serial_ports = int(params.get("virtio_serial_ports"))
+    for i in range(1, num_chardev):
+        params["extra_chardevs"] += ' channel%d' % i
+        serial_name = 'port%d' % i
+        params['extra_serials'] = '%s %s' % (params.get('extra_serials', ''),
+                                             serial_name)
+        params['serial_type_%s' % serial_name] = "virtserialport"
+    char_devices = add_chardev(vm, params)
+    serials = params.objects('extra_serials')
+    buses, serial_devices = get_buses_and_serial_devices(
+        vm, params, char_devices, serials)
+    vm.devices.simple_hotplug(buses[0], vm.monitor)
+    for i in range(0, num_chardev):
+        vm.devices.simple_hotplug(char_devices[i], vm.monitor)
+        if i < num_serial_ports:
+            vm.devices.simple_hotplug(serial_devices[i], vm.monitor)
+    for device in serial_devices:
+        add_virtio_ports_to_vm(vm, params, device)
+    if os_type == "windows":
+        driver_name = params["driver_name"]
+        session = vm.wait_for_login()
+        session = utils_test.qemu.windrv_check_running_verifier(
+            session, vm, test, driver_name)
+    run_bg_test()
+    try:
+        out1 = vm.devices.simple_hotplug(serial_devices[0], vm.monitor)
+        out2 = vm.devices.simple_hotplug(char_devices[0], vm.monitor)
+        if out1[1] is True:
+            msg = ("The device %s shoudn't be hotplugged successfully"
+                   % serial_devices[0])
+            test.fail(msg)
+        elif out2[1] is True:
+            msg = ("The device %s shoudn't be hotplugged successfully"
+                   % char_devices[0])
+            test.fail(msg)
+    except DeviceHotplugError:
```

Comment:
Normally, `vm.devices.simple_hotplug(serial_devices[0], vm.monitor)` will get this exception, will skip the next step `simple_hotplug(char_devices[0], vm.monitor)`.   Miss one check point

Reply:
Thanks and update